zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Service.java

```diff
@@ -0,0 +1,85 @@
+package com.github.onsdigital.zebedee.api;
+
+import com.github.davidcarboni.cryptolite.Random;
+import com.github.davidcarboni.restolino.framework.Api;
+import com.github.onsdigital.zebedee.exceptions.NotFoundException;
+import com.github.onsdigital.zebedee.exceptions.UnauthorizedException;
+import com.github.onsdigital.zebedee.json.PermissionDefinition;
+import com.github.onsdigital.zebedee.model.ServiceAccount;
+import com.github.onsdigital.zebedee.permissions.service.PermissionsService;
+import com.github.onsdigital.zebedee.service.ServiceStore;
+import com.github.onsdigital.zebedee.service.ServiceStoreImpl;
+import com.github.onsdigital.zebedee.session.model.Session;
+import com.github.onsdigital.zebedee.session.service.SessionsService;
+import org.apache.commons.lang3.BooleanUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.POST;
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import static com.github.onsdigital.zebedee.logging.ZebedeeLogBuilder.logInfo;
+
+@Api
+public class Service {
+
+    private Supplier<String> randomIdGenerator = () -> Random.id();
+
+    private ServiceStore serviceStore;
+
+    private SessionsService sessionsService;
+
+    private PermissionsService permissionsService;
+
+    @POST
+    public ServiceAccountWithToken createService(HttpServletRequest request, HttpServletResponse response) throws IOException, NotFoundException, UnauthorizedException {
+        final Session session = getSessionsService().get(request);
```

Comment:
I think this can return null - which would cause a null pointer on session.getEmail(), can you add a check for this and return the appropriate response.